### PR TITLE
Email domain mapping frontend

### PR DESCRIPTION
**Pull request by cvializ**
@andrewconner @camhashemi @eishay @JRuff42 
screen shot 2015-11-16 at 7 58 28 pm
screen shot 2015-11-16 at 7 58 45 pm
screen shot 2015-11-16 at 7 58 39 pm

### CODE REVIEW

File: kifi-backend/modules/shoebox/angular/src/common/directives/proxy.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      terminal: true,
+      priority: 1000000,
+      replace: true,
+      template: '<span></span>',
+      link: function (scope, element, attrs) {
+        directiveProxyService(attrs.proxy, scope, element, attrs, ['proxy']);
+      }
+    };
+  }
+]);
```
Comment (cvializ) [inline kifi-backend/modules/shoebox/angular/src/common/directives/proxy.js:48]: so yeah this is kind of weird, but it has a syntactically clean and intuitive way to variably apply a directive, which maintains the elegance of the teamSettings.js data-structure

File: kifi-backend/modules/shoebox/angular/src/profile/profileWidget.tpl.html [hunk 2]
```diff
@@ -111,7 +111,7 @@
   <div ng-if="::shouldShowCreateTeam()" class="kf-pw-create-org" >
     <div class="kf-pw-create-org-wrapper kf-link-parent" ng-click="createTeam()">
       <span class="svg-create-library kf-pw-create-org-icon"></span>
-      <button class="kf-pw-create-org-label kf-link">Create a team</button>
+      <button class="kf-pw-create-org-label kf-link" ng-bind="'Create a team' + (companyName ? ' for ' + companyName : '')"></button>
```
Comment (cvializ) [inline kifi-backend/modules/shoebox/angular/src/profile/profileWidget.tpl.html:114]: bonus https://app.asana.com/0/39814736029951/60353532458199

File: kifi-backend/modules/shoebox/angular/src/common/directives/proxy.js (hunk 1 above)
Comment (camhashemi) [inline kifi-backend/modules/shoebox/angular/src/common/directives/proxy.js:48]: the elegance of the `teamSettings.js` data structure is of utmost importance

Comment (camhashemi) [inline kifi-backend/modules/shoebox/angular/src/common/directives/proxy.js:48]: plus it's actually nice

File: kifi-backend/modules/shoebox/angular/src/teamSettings/teamEmailMapping.tpl.html [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    <span class="kf-team-email-mapping-toggle">
+      <span ng-if="!open">+</span>
+      <span ng-if="open">-</span>
+    </span>
+    <span class="kf-link">Add email domains</span>
+  </button>
+  <div ng-show="open">
+    <p class="kf-team-settings-desc">
+      Add company email domains that will enable your coworkers to automatically join this team.
+      When a user vertifies their email address with Kifi, they will automatically be added to the team.
```
Comment (camhashemi) [inline kifi-backend/modules/shoebox/angular/src/teamSettings/teamEmailMapping.tpl.html]: verifies*

Comment (cvializ) [inline kifi-backend/modules/shoebox/angular/src/teamSettings/teamEmailMapping.tpl.html]: thank you!

**Pull request merged**